### PR TITLE
Don't block an offload thread when the client's SecurityPolicy is async

### PR DESCRIPTION
**Pull request by jdcormie**
https://github.com/grpc/grpc-java/issues/10566

### CODE REVIEW

File: binder/src/main/java/io/grpc/binder/internal/BinderTransport.java [hunk 1]
```diff
@@ -743,57 +746,63 @@ void notifyTerminated() {
...
     }
 
-    private void checkSecurityPolicy(IBinder binder) {
-      Status authorization;
-      Integer remoteUid;
-      synchronized (this) {
-        remoteUid = attributes.get(REMOTE_UID);
-      }
-      if (remoteUid == null) {
-        authorization = Status.UNAUTHENTICATED.withDescription("No remote UID available");
```
Comment (kannanjgithub) [inline binder/src/main/java/io/grpc/binder/internal/BinderTransport.java:770]: The new changes are not checking for this possibility of Binder.getCallinguid() being null.

Comment (jdcormie) [reply binder/src/main/java/io/grpc/binder/internal/BinderTransport.java:770]: [Binder.getCallinguid() ](https://developer.android.com/reference/android/os/Binder#getCallingUid()) returns int so null isn't actually possible. I believe the old code was just being defensive.

**Review by kannanjgithub — APPROVED** [review]

File: binder/src/main/java/io/grpc/binder/internal/BinderTransport.java [hunk 2]
```diff
@@ -755,45 +758,52 @@ protected void handleSetupTransport(Parcel parcel) {
...
+            clientTransportListener.transportReady();
+            if (readyTimeoutFuture != null) {
+              readyTimeoutFuture.cancel(false);
+              readyTimeoutFuture = null;
             }
           }
         }
       }
     }
+    private synchronized void handleAuthResult(Throwable t) {
```
Comment (ejona86) [inline binder/src/main/java/io/grpc/binder/internal/BinderTransport.java:800]: nit: add empty line before method

**Review by ejona86 — APPROVED** [review]

File: binder/src/main/java/io/grpc/binder/internal/BinderTransport.java (hunk 2 above)
Comment (jdcormie) [reply binder/src/main/java/io/grpc/binder/internal/BinderTransport.java:801]: done

**Comment by jdcormie** [comment]
cc @mateusazis

**Pull request merged**